src/intervals/conversion.jl

```diff
@@ -12,23 +12,32 @@ promote_rule(::Type{BigFloat}, ::Type{Interval{T}}) where T<:Real =
     Interval{promote_type(T, BigFloat)}
 
 
+# convert methods:
+convert(::Type{Interval{T}}, x) where {T} = closure(Interval{T}, x)
+convert(::Type{Interval{T}}, x::T) where {T} = Interval{T}(x)
+convert(::Type{Interval{T}}, x::Interval{T}) where {T} = x
+convert(::Type{Interval{T}}, x::Interval) where {T} = closure(Interval{T}, x)
+
+convert(::Type{Interval}, x) = closure(Interval, x)
+convert(::Type{Interval}, x::Real) = (T = typeof(float(x)); convert(Interval{T}, x))
+convert(::Type{Interval}, x::Interval) = x
 
-# Floating point intervals:
+# Integer intervals:
+closure(::Type{Interval{T}}, x::T) where {T<:Integer} = Interval{T}(x)
 
-convert(::Type{Interval{T}}, x::AbstractString) where T<:AbstractFloat =
+# Floating point intervals:
+closure(::Type{Interval{T}}, x::AbstractString) where T<:AbstractFloat =
     parse(Interval{T}, x)
 
-convert(::Type{Interval}, x::AbstractString) = convert(Interval{Float64}, x)
-
-function convert(::Type{Interval{T}}, x::S) where {T<:AbstractFloat, S<:Real}
+function closure(::Type{Interval{T}}, x::S) where {T<:AbstractFloat, S<:Real}
     isinf(x) && return wideinterval(T(x))
 
     Interval{T}( T(x, RoundDown), T(x, RoundUp) )
     # the rounding up could be done as nextfloat of the rounded down one?
     # use @round_up and @round_down here?
 end
 
-function convert(::Type{Interval{T}}, x::S) where {T<:AbstractFloat, S<:AbstractFloat}
+function closure(::Type{Interval{T}}, x::S) where {T<:AbstractFloat, S<:AbstractFloat}
     isinf(x) && return wideinterval(x)#Interval{T}(prevfloat(T(x)), nextfloat(T(x)))
```

Comment:
A small remark, which is not due to the changes you propose (perhaps it is me the one to be blamed!), but this is a good chance to correct it: This method is not type stable. To be type stable, I think it should return `wideinterval(T(x))`.

An example that shows the problem is:
```julia
julia> typeof( IntervalArithmetic.closure(Interval{BigFloat}, Inf) )
IntervalArithmetic.Interval{Float64}

julia> typeof( IntervalArithmetic.closure(Interval{BigFloat}, 0.1) )
IntervalArithmetic.Interval{BigFloat}
```

Reply:
As you prefer...